src/shogun/multiclass/ShareBoost.cpp

```diff
@@ -43,6 +43,7 @@ SGVector<int32_t> CShareBoost::get_activeset()
 
 bool CShareBoost::train_machine(CFeatures* data)
 {
+	CDenseFeatures<float64_t> *fea = dynamic_cast<CDenseFeatures<float64_t>*>(m_features);
```

Comment:
@c4goldsw @vigsterkr 
Crhis, here is another one of those float64 casts, which shouldnt be there.
@OXPHOS just fixed a minor thing here, but we should systematically get rid of such casts everywhere. Can probably grep for them and have a list somewhere